plugins/advisors/black-duck/src/funTest/kotlin/ResponseCachingComponentServiceClient.kt

```diff
@@ -44,7 +44,8 @@ internal class ResponseCachingComponentServiceClient(
     // The Black Duck library uses GSON to serialize its POJOs. So use GSON, too, because this is the simplest option.
     private val gson = GsonBuilder().setPrettyPrinting().create()
 
-    private val cache = if (overrideFile.isFile) {
+    private val overrideFileExisted = overrideFile.isFile
```

Comment:
This file contains all the data needed for the fun tests.
These are multiple test cases. So, this change would lead to the behavior that only data for the first run case is written out IIUC, while it would be required to be able to re-create the file so that it contains the data needed by all tests.

Reply:
> So, this change would lead to the behavior that only data for the first run case is written out IIUC

I believe that's not correct: `ResponseCachingComponentServiceClient` in only created once at the start of `BlackDuckFunTest`, so also this property is only initialized once at the start of the test, and saves whether the file existed at the start of the test. It doe snot redo the existence check between tests. At least that was my intention when implementing this.

> Proposal: Change the logic to write out the file if and only if a server / credentials are specified.

That's exactly the existing implementation: `delegate` is only non-null if `serverUrl != null && apiToken != null`.

Reply:
> That's exactly the existing implementation: `delegate` is only non-null if `serverUrl != null && apiToken != null`.

I see. So can we keep the existing implementation, because in our setup it does not leave the file tree "dirty" ?
(maye, I do not get what problem you are trying to solve for what reason)

Reply:
> maye, I do not get what problem you are trying to solve for what reason

I thought it's a requirement (from your side) that it should also be possible to run this test against a real server when (later, once #10025 is implemented) running the test from a JAR. However, when running the test from a JAR, the file must not be rewritten. So I refined the logic to not *always* rewrite the file when running against a server, but only if the overwrite file was missing before running the test. As this is only possible during development when *not* running from a JAR, I do not see how this change makes updating the file less convenient.

Reply:
> I do not see how this change makes updating the file less convenient.

Incremental updates are no more as convenient as they were before, or not?

Reply:
What exactly *is* an "incremental update" in your view here? Are you referring to successive updates by the individual tests in the spec, or by successive runs of the whole spec?

Reply:
I guess you meant to *update* the committed `recorded-responses.json` with responses from the server?

Reply:
Ok. Then how about if I change the logic so that the file is never updated when running from a JAR, even when running against a server, and for all other cases I keep the behavior as-is?

Reply:
> Ok. Then how about if I change the logic so that the file is never updated when running from a JAR, even when running against a server, and for all other cases I keep the behavior as-is?

This makes sense to me.

Reply:
I've now decided to integrate this into https://github.com/oss-review-toolkit/ort/pull/10294, please have a look.